roles/README.adoc

```diff
@@ -399,6 +399,52 @@ described in the
 https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html[Ansible docs].
 ====
 
+=== Don't use host group names or at least make them a parameter
+[%collapsible]
+====
+Explanations::
+It is relatively common to use (inventory) group names in roles:
++
+* either to loop through the hosts in the group, generally in a cluster context
+* or to validate that a host is in a specific group
++
+Use instead a variable (a list variable in the first case), or at least make thegroup's name a parameter of your role.
+You can always set the variable at group level to avoid repetitions.
+
+Rationale::
+Groups are a feature of the data in your inventory, meaning that you mingle data with code when you use those groups in your code.
+You should better rely on the inventory parsing process to provide your code with the variables it needs, not enforcing a specific structure of the inventory.
+Not all inventory sources are flexible enough to provide exactly the expected group name.
+Even more importantly, in a cluster context for example, if the group name is fixed, you can't describe (and hence automate) more than one cluster in each inventory.
```

Comment:
```suggestion
Even more importantly, in a cluster context for example, if the group name is fixed, you can't describe (and hence automate) more than one cluster in each inventory becasue you cannot define multiple groups with the same name without having multiple inventories.
```

Reply:
That's IMHO an obvious statement (and there is a typo).